### PR TITLE
Remove element-scheme support for incompressible solver

### PR DESCRIPTION
**Pull request by sayerhs**
See #829 for deprecation details. 

Additional changes:

- Switch default scheme to `edges` in Realm. Throw error if the scheme is not `edges` or `matrix-free`
- Convert all overset and OpenFAST actuator line tests to `use_edges: true`
- Switch `ActLineSimple` to edge-based scheme (superseded by `ActLineSimpleNGP` anyway)
- Remove regression tests that used element-based schemes (restore if needed on a case-by-case basis)

Other details/issues:

- Currently, checks are performed within each `EquationSystem`. Clean-up of `SolutionOptions` still pending.
- Some element kernels and legacy, non-NGP element algorithms are still retained because they are used by `HCEQS` and `PNGEQS`. Should either convert to NGP versions or remove these EQS to allow for removal of these kernels. 
- DG interface retained with edge-based scheme for sliding mesh

### CODE REVIEW

**Comment by sayerhs** [comment]
@lawrenceccheung @tonyinme I added you both so that you can check the changes NREL5MW actuator tests and the `ActLineSimple` tests.

**Review by tonyinme — APPROVED** [review]
Hey @sayerhs, this looks good to me. There was no real reason for us not to use edges before. As long as the tests pass, this should all be good.

**Comment by psakievich** [comment]
I ran this on one of the compute servers using GCC 7.2 and saw the following related failures:
```
         25 - ActLineSimple (Failed)
         39 - nrel5MWactuatorLineAnisoGauss (Failed)
         40 - nrel5MWactuatorLineFllc (Failed)
         41 - nrel5MWactuatorDisk (Failed)
         42 - nrel5MWadvActLine (Failed)
         43 - oversetSphereTIOGA (Failed)
         44 - oversetRotCylinder (Failed)
```
These were all elem tests that @sayerhs noted were converted to the edge scheme so no surprises to see the diffs.

File: reg_tests/CTestList.cmake [hunk 1]
```diff
@@ -121,63 +121,22 @@ if(NOT ENABLE_CUDA)
...
   add_test_r(edgeHybridFluids 8)
   add_test_r(ekmanSpiral 4)
-  add_test_r(ekmanSpiralConsolidated 4)
-  add_test_r_inp(elemBackStepLRSST 4)
-  add_test_r(elemClosedDomain 2)
-  add_test_r(elemHybridFluids 8)
-  add_test_r(elemHybridFluidsShift 8)
-  add_test_r(femHC 2)
-  add_test_r(femHCGL 2)
-  add_test_r(heatedBackStep 4)
```
Comment (psakievich) [inline reg_tests/CTestList.cmake:153]: I'm wondering if we should convert a few more of these tests to be edge scheme. Browsing through, I'm thinking about:

- `closedDomain`
- at least one of the `backStep`'s
- `milestoneRun`
- `variableDensNonIso` (this might be better to swap the with the NonUniform version of the test which uses mixture fraction. IIRC we're going to remove mixture fraction at some point as well)

File: reg_tests/CTestList.cmake [hunk 2]
```diff
@@ -242,8 +199,6 @@ if(NOT ENABLE_CUDA)
   #=============================================================================
   # Performance tests
   #=============================================================================
-  add_test_p(uqSlidingMeshDG 8)
```
Comment (psakievich) [inline reg_tests/CTestList.cmake:245]: We can also remove the performance test macros from the top of this file unless we want to replace them with more targeted wind energy edge variants. I'm open to either option.

**Review by psakievich — CHANGES_REQUESTED** [review]
@sayerhs thanks for doing this. Looks really good.  I noticed the `ductWedge` test is an edge test that I don't think should be removed. That's the only change I'm requesting.  I think it is worth discussing converting a few more tests from elem to edge to make sure we have adequate coverage.  But I don't think those conversions should hold up this PR.  We can certainly address them in a follow on PR. Looking forward to trimming the mesh repo once this goes through :)

File: reg_tests/CTestList.cmake (hunk 2 above)
Comment (sayerhs) [reply reg_tests/CTestList.cmake:245]: @psakievich I was hoping to add wind-energy specific performance tests here. The empty section header was meant as reminder for whoever that ends up adding these tests later.

File: reg_tests/CTestList.cmake (hunk 1 above)
Comment (sayerhs) [reply reg_tests/CTestList.cmake:153]: When we reinstate these tests, we should rename tests like `milestoneRun` with something more relatable

Comment (psakievich) [reply reg_tests/CTestList.cmake:153]: Agreed

File: reg_tests/CTestList.cmake (hunk 2 above)
Comment (psakievich) [reply reg_tests/CTestList.cmake:245]: Sounds good to me

**Review by marchdf — APPROVED** [review]
three thumbs-up!

**Review by psakievich — APPROVED** [review]
After speaking with @sayerhs we agreed there isn't anything special in the `ductWedge` case to make it a critical case to preserve in the regression test suite and hold back this PR.
